src/mappings/bank/supply.ts

```diff
@@ -0,0 +1,47 @@
+import type {
+  Coin,
+  CosmosBlock,
+} from "@subql/types-cosmos";
+import type { QueryTotalSupplyResponse } from "cosmjs-types/cosmos/bank/v1beta1/query";
+import { Supply } from "../../types";
+import { stringify } from "../utils";
+
+export const getSupplyRecord = (supply: Coin, block: CosmosBlock): Supply => Supply.create({
+  id: `${supply.denom}@${block.block.id}`,
```

Comment:
Wdyt about factoring this out to a `getSupplyId` function, similar to the others that we have for message and event ids?